pkg/spdk/types.go

```diff
@@ -1,16 +1,56 @@
 package spdk
 
-import "fmt"
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/longhorn/longhorn-spdk-engine/pkg/client"
+	"github.com/longhorn/longhorn-spdk-engine/pkg/types"
+	"github.com/longhorn/longhorn-spdk-engine/pkg/util"
+)
 
 const (
 	DiskTypeFilesystem = "filesystem"
 	DiskTypeBlock      = "block"
+
+	ReplicaRebuildingLvolSuffix  = "rebuilding"
+	RebuildingSnapshotNamePrefix = "rebuild"
 )
 
+func GetReplicaSnapshotLvolNamePrefix(replicaName string) string {
+	return fmt.Sprintf("%s-snap-", replicaName)
+}
+
 func GetReplicaSnapshotLvolName(replicaName, snapshotName string) string {
-	return fmt.Sprintf("%s-snaps-%s", replicaName, snapshotName)
+	return fmt.Sprintf("%s%s", GetReplicaSnapshotLvolNamePrefix(replicaName), snapshotName)
+}
```

Comment:
The max length of a lvol's name is 64 bytes.
If a name generated by `GetReplicaSnapshotLvolName` exceeds the number, the snapshot operation will fail.

Reply:
Can you update the max length (`SPDK_LVOL_NAME_MAX`) to a bigger number?
cc @DamiaSan

Reply:
Handled in https://github.com/longhorn/spdk/pull/11

Reply:
Not sure if upstream would accept this change... Do we need to truncate `replicaName` if it's too long?

Reply:
> Do we need to truncate replicaName if it's too long?

I'm fine with the idea.
However, 64 bytes is really short. The NAME_MAX on Linux is 255 ([ref](https://elixir.bootlin.com/linux/latest/source/include/uapi/linux/limits.h#L12)). Increasing the number is reasonable to me.

@DamiaSan Can you help check the change with SPDK maintainers? Thank you.

Reply:
Let's do a workaround here first by truncating if no extra concerns, but at the same, see if 255 name length is feasible for upstream. @shuo-wu @derekbit ?